propagation.go

```diff
@@ -57,53 +59,46 @@ type Extractor interface {
 	Extract(carrier interface{}) (SpanContext, error)
 }
 
-type textMapPropagator struct {
-	tracer      *tracer
-	encodeValue func(string) string
-	decodeValue func(string) string
+type coder interface {
+	Encode(string) string
+	Decode(string) string
 }
 
-func newTextMapPropagator(tracer *tracer) *textMapPropagator {
-	return &textMapPropagator{
-		tracer: tracer,
-		encodeValue: func(val string) string {
-			return val
-		},
-		decodeValue: func(val string) string {
-			return val
-		},
-	}
+type textCoder struct{}
+
+func (p *textCoder) Encode(val string) string {
```

Comment:
(not too relevant since this code should go away, but a nit): the receiver can be plain value, not a pointer.